gc/base/MemorySubSpaceUniSpace.cpp

```diff
@@ -257,10 +257,16 @@ MM_MemorySubSpaceUniSpace::timeForHeapContract(MM_EnvironmentBase *env, MM_Alloc
 		return false;
 	}
 
-	/* Don't shrink if we have not met the allocation request
-	 * ..we will be expanding soon if possible anyway
+	/* Don't shrink if this is implicit aggressive GC or we have not met the allocation request,
+	 * we will be expanding soon if possible anyway
 	 */
 	if (allocDescription) {
+		if (env->_cycleState->_gcCode.isImplicitAggressiveGC()) {
+			Trc_MM_MemorySubSpaceUniSpace_timeForHeapContract_Exit8(env->getLanguageVMThread());
+			_contractionSize = 0;
```

Comment:
The exit tracepoint should be last operation before the return statement. Is there a reason for setting `_contractionSize` after the exit tracepoint? This is also done at other places in this function.
```suggestion
			_contractionSize = 0;
			Trc_MM_MemorySubSpaceUniSpace_timeForHeapContract_Exit8(env->getLanguageVMThread());
```

Reply:
We missed one on lines 343-344.